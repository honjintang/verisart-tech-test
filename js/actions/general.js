export const saveAnswer = (id, value) => ({
    type: 'SAVE_ANSWER',
    payload: value,
    meta: id,
})