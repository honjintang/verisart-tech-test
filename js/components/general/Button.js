import React from 'react'

const Button = ({ text, onClickFn }) => (
    <button className="loginButton" onClick={onClickFn}>{text}</button>
)

export default Button