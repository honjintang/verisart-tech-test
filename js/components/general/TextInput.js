import React from 'react'
import { connect } from 'react-redux'

const mapStateToProps = state => ({
    error: state.error
})

const TextInput = ({ type, name, placeholder, error }) => (
    <div className="textInputWrapper" >
        <input type={type} className={`${error && "errorInput"} textInput`} name={name} placeholder={placeholder} />
        {error && <div className="errorFieldMessage">{error.message}</div>}
    </div>
)

export default connect(mapStateToProps)(TextInput)