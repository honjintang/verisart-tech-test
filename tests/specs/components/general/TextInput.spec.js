import React from 'react'
import { shallow } from 'enzyme'
import { TextInput } from '~/components/general/TextInput'

const mock_saveAnswer = jest.fn()
const mock_clearAnswer = jest.fn()

describe('components/display/TextInput', () => {
    it('matches snapshot when no error', () => {
        const component = shallow(<TextInput
             _saveAnswer={mock_saveAnswer} 
             _clearAnswer={mock_clearAnswer} 
             type='email'
             name='email'
             answer='lalala'
             placeholder='mockplaceholder'
              />)
        expect(component).toMatchSnapshot()
    })

    it('matches snapshot when error', () => {
        const component = shallow(<TextInput
             _saveAnswer={mock_saveAnswer} 
             _clearAnswer={mock_clearAnswer} 
             type='email'
             name='email'
             answer='lalala'
             placeholder='mockplaceholder'
             error={{ message: 'this is an error', id: 'email'}}
              />)
        expect(component).toMatchSnapshot()
    })
})