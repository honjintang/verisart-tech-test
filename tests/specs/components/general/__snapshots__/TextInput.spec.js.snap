// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/TextInput matches snapshot when error 1`] = `
<div
  className="textInputWrapper"
>
  <input
    className="errorInput textInput"
    name="email"
    onChange={[Function]}
    placeholder="mockplaceholder"
    type="email"
    value="lalala"
  />
  <div
    className="errorFieldMessage"
  >
    this is an error
  </div>
</div>
`;

exports[`components/display/TextInput matches snapshot when no error 1`] = `
<div
  className="textInputWrapper"
>
  <input
    className="undefined textInput"
    name="email"
    onChange={[Function]}
    placeholder="mockplaceholder"
    type="email"
    value="lalala"
  />
</div>
`;
